smac/epm/rfr_imputator.py

```diff
@@ -106,7 +108,7 @@ def impute(self, censored_X: np.ndarray, censored_y: np.ndarray,
         imputed_y = None  # define this, if imputation fails
 
         # Define variables
-        y = None
+        y = np.empty((0, ))  # This only defines the type, the actual value will not be used later on.
```

Comment:
Why is it not type: Optional[ndarray]?
Also I just realized for a similar case in https://github.com/automl/SMAC3/blob/a1d6c4d8919b9ce3a23be322835e8367162b8d55/smac/epm/gaussian_process.py#L89
it is typed as type: ndarray

Reply:
Because then it would complain that you cannot perform operations on objects of type `None`. In general, handling `None` correctly is a bit harder with type annotations...

Reply:
I see. Ok then I'm fine with `np.empty((0,  ))`